### ISSUE TITLE
[iOS 14][WKWebView] WKWebView is not loading iFrame and is not accepting Cookies

### ISSUE THREAD

**Issue by bnoffer** [xamarin/xamarin-macios#9831]
### Steps to Reproduce

1. Open Page with WKWebView pointing to page with iFrame
2. Wait to see the iFrame load
3. Nothing happens

### Expected Behavior

The iFrame content should load.

### Actual Behavior

In our app we use a iFrame to point to a Javascript generated page provided by one of our partner companies. For some reason neither cookies seem to be working nor the Javascript processing. In the debug output of the App I get the following:

```
2020-10-09 10:12:42.283182+0200 MyApp.iOS[44281:3202438] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:12:42.283374+0200 MyApp.iOS[44281:3202438] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:12:43.042932+0200 MyApp.iOS[44281:3202438] [TraitCollection] Class CKBrowserSwitcherViewController overrides the -traitCollection getter, which is not supported. If you're trying to override traits, you must use the appropriate API.
2020-10-09 10:13:04.780285+0200 MyApp.iOS[44281:3203141] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:13:04.780567+0200 MyApp.iOS[44281:3203141] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:13:04.788062+0200 MyApp.iOS[44281:3202438] [unspecified] container_system_group_path_for_identifier: error = ((container_error_t)98) NOT_CODESIGNED
2020-10-09 10:13:04.788180+0200 MyApp.iOS[44281:3202438] [MC] Error getting system group container for systemgroup.com.apple.configurationprofiles: 98
2020-10-09 10:13:04.788387+0200 MyApp.iOS[44281:3202438] [MC] Failed to get profile system group container path. Overriding with expected path: /Users/ban/Library/Developer/CoreSimulator/Devices/2862C1BD-17F7-425A-A883-8B0CF601E088/data/Containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2020-10-09 10:13:04.791846+0200 MyApp.iOS[44281:3202438] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:13:04.792034+0200 MyApp.iOS[44281:3202438] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:13:04.792231+0200 MyApp.iOS[44281:3202438] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:13:04.792391+0200 MyApp.iOS[44281:3202438] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:13:04.862517+0200 MyApp.iOS[44281:3202438] WF: === Starting WebFilter logging for process MyApp.iOS
2020-10-09 10:13:04.862729+0200 MyApp.iOS[44281:3202438] WF: _WebFilterIsActive returning: NO
2020-10-09 10:13:04.876457+0200 MyApp.iOS[44281:3202438] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:13:04.876665+0200 MyApp.iOS[44281:3202438] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:13:04.920782+0200 MyApp.iOS[44281:3203757] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=44281
2020-10-09 10:13:04.920956+0200 MyApp.iOS[44281:3203757] SecTaskCopyDebugDescription: MyApp.iOS[44281]/0#-1 LF=0
2020-10-09 10:13:04.921318+0200 MyApp.iOS[44281:3203757] [] nw_protocol_get_quic_image_block_invoke dlopen libquic failed
2020-10-09 10:13:05.431102+0200 MyApp.iOS[44281:3202438] WF: _WebFilterIsActive returning: NO
2020-10-09 10:13:05.447267+0200 MyApp.iOS[44281:3202438] [unspecified] container_system_group_path_for_identifier: error = ((container_error_t)98) NOT_CODESIGNED
```
This issue only occurs since building against iOS 14 and only applies to iOS 14. Running the same build on iOS 13 and lower the content is loaded properly. We also already tried applying the App-Bound Domains feature (see: https://webkit.org/blog/10882/app-bound-domains/ ) without any luck.

I tried to research some of the messages displaye in the log without a real hint on what is going wrong.

### Environment

```
Visual Studio Enterprise 2019 for Mac
Version 8.7.8 (build 4)
Installation UUID: 8fa54a85-0052-45ac-a643-0f7597976eff
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000093

Mono Framework MDK
Runtime:
	Mono 6.12.0.93 (2020-02/620cf538206) (64-bit)
	Package version: 612000093

Roslyn (Language Service)
3.7.0-6.20427.1+18ede13943b0bfae1b44ef078b2f3923159bcd32

NuGet
Version: 5.7.0.6702

.NET Core SDK
SDK: /usr/local/share/dotnet/sdk/3.1.402/Sdks
SDK Versions:
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	3.0.100-rc1-014190
	3.0.100-preview8-013656
	2.2.203
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Sdks

.NET Core Runtime
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	3.0.0-rc1-19456-20
	3.0.0-preview8-28405-07
	2.2.4
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.0.5

Xamarin.Profiler
Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode 12.0.1 (17220)
Build 12A7300

Xamarin.Mac
Version: 6.20.2.2 (Visual Studio Enterprise)
Hash: 817b6f72a
Branch: d16-7
Build date: 2020-07-18 18:44:59-0400

Xamarin.iOS
Version: 14.0.0.0 (Visual Studio Enterprise)
Hash: 7ec3751a1
Branch: xcode12
Build date: 2020-09-16 11:33:15-0400

Xamarin Designer
Version: 16.7.0.495
Hash: 03d50a221
Branch: remotes/origin/d16-7-vsmac
Build date: 2020-08-28 13:12:52 UTC

Xamarin.Android
Version: 11.0.2.0 (Visual Studio Enterprise)
Commit: xamarin-android/d16-7/025fde9
Android SDK: /Users/ban/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		6.0 (API level 23)
		7.0 (API level 24)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 29.0.3

Build Information: 
Mono: 83105ba
Java.Interop: xamarin/java.interop/d16-7@1f3388a
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-7@017078f

Microsoft OpenJDK for Mobile
Java SDK: /Users/ban/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 16.7.0.13
Hash: 8380518
Branch: remotes/origin/d16-7~2
Build date: 2020-09-16 05:12:24 UTC

Android Device Manager
Version: 16.7.0.24
Hash: bb090a3
Branch: remotes/origin/d16-7
Build date: 2020-09-16 05:12:46 UTC

Build Information
Release ID: 807080004
Git revision: 9ea7bef96d65cdc3f4288014a799026ccb1993bc
Build date: 2020-09-16 17:22:54-04
Build branch: release-8.7
Xamarin extensions: 9ea7bef96d65cdc3f4288014a799026ccb1993bc

Operating System
Mac OS X 10.15.6
Darwin 19.6.0 Darwin Kernel Version 19.6.0
    Thu Jun 18 20:49:00 PDT 2020
    root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64

Enabled user installed extensions
DeepClean 1.2.5

```

**Comment by chamons** [comment]
Thank you for your feedback!

This line:

```
2020-10-09 10:13:05.447267+0200 MyApp.iOS[44281:3202438] [unspecified] container_system_group_path_for_identifier: error = ((container_error_t)98) NOT_CODESIGNED
```

suggests that code signing or entitlements may be the issue.

For us to investigate this further, could you please provide a test case (to reproduce).

**Comment by bnoffer** [comment]
The issue seems to be resolved with 14.4.0.9 beta release.

I can not provide you with a proper sample, due to legal obligations of the whole context.
